core/src/test/java/com/google/cloud/sql/core/ConnectorTest.java

```diff
@@ -0,0 +1,131 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.sql.core;
+
+import static com.google.common.truth.Truth.assertThat;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.junit.Assert.fail;
+
+import com.google.cloud.sql.ConnectionConfig;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.Socket;
+import java.time.Duration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ConnectorTest extends CloudSqlCoreTestingBase {
+  ListeningScheduledExecutorService defaultExecutor;
+  private final long TEST_MAX_REFRESH_MS = 5000L;
+
+  @Before
+  public void setUp() throws Exception {
+    super.setup();
+    defaultExecutor = InternalConnectorRegistry.getDefaultExecutor();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    defaultExecutor.shutdownNow();
+  }
+
+  @Test
+  public void create_throwsErrorForInvalidInstanceName() throws IOException {
+    ConnectionConfig config =
+        new ConnectionConfig.Builder()
+            .withCloudSqlInstance("myProject")
+            .withIpTypes("PRIMARY")
+            .build();
+
+    ConnectionConfig config2 =
+        new ConnectionConfig.Builder()
+            .withCloudSqlInstance("myProject:myRegion")
+            .withIpTypes("PRIMARY")
+            .build();
+
+    Connector c = newConnector(config, 3307);
+    try {
+      c.connect(config);
+      fail();
+    } catch (IllegalArgumentException e) {
+      assertThat(e).hasMessageThat().contains("Cloud SQL connection name is invalid");
+    }
+
+    try {
+      c.connect(config2);
+      fail();
+    } catch (IllegalArgumentException e) {
+      assertThat(e).hasMessageThat().contains("Cloud SQL connection name is invalid");
+    }
+  }
+
+  /**
+   * Start an SSL server on the private IP, and verifies that specifying a preference for private IP
+   * results in a connection to the private IP.
+   */
+  @Test
+  public void create_successfulPrivateConnection() throws IOException, InterruptedException {
+    FakeSslServer sslServer = new FakeSslServer();
+    ConnectionConfig config =
+        new ConnectionConfig.Builder()
+            .withCloudSqlInstance("myProject:myRegion:myInstance")
+            .withIpTypes("PRIVATE")
+            .build();
+
+    int port = sslServer.start(PRIVATE_IP);
+
+    Connector connector = newConnector(config, port);
+
+    Socket socket = connector.connect(config);
+
+    assertThat(readLine(socket)).isEqualTo(SERVER_MESSAGE);
+  }
+
+  // TODO add and adapt the rest of the relevant testcases from CoreSocketFactoryTest:
```

Comment:
Do we have an issue for this? I'd like to use GitHub issue to track non-trivial TODOs instead of comments (which will be forgotten).

Reply:
Good idea. See #1656.